src/vstest.console/Processors/TestAdapterPathArgumentProcessor.cs

```diff
@@ -161,44 +166,39 @@ public void Initialize(string argument)
 
             try
             {
+                var testAdapterPaths = new List<string>();
+                var testAdapterFullPaths = new List<string>();
+                
                 // VSTS task add double quotes around TestAdapterpath. For example if user has given TestAdapter path C:\temp,
                 // Then VSTS task will add TestAdapterPath as "/TestAdapterPath:\"C:\Temp\"".
                 // Remove leading and trailing ' " ' chars...
                 argument = argument.Trim().Trim(new char[] { '\"' });
-                customAdaptersPath = Path.GetFullPath(argument);
-                if (!fileHelper.DirectoryExists(customAdaptersPath))
-                {
-                    throw new DirectoryNotFoundException(CommandLineResources.TestAdapterPathDoesNotExist);
-                }
 
                 // Get testadapter paths from RunSettings.
                 var testAdapterPathsInRunSettings = this.runSettingsManager.QueryRunSettingsNode("RunConfiguration.TestAdaptersPaths");
 
                 if (!string.IsNullOrWhiteSpace(testAdapterPathsInRunSettings))
                 {
-                    var testAdapterFullPaths = new List<string>();
-                    var testAdapterPathsInRunSettingsArray = testAdapterPathsInRunSettings.Split(
-                        new[] { ';' },
-                        StringSplitOptions.RemoveEmptyEntries);
+                    testAdapterPaths.AddRange(SplitPaths(testAdapterPathsInRunSettings));
+                }
+                
+                testAdapterPaths.AddRange(SplitPaths(argument));
+                
+                foreach (var testadapterPath in testAdapterPaths)
+                {
+                    var testAdapterFullPath = Path.GetFullPath(testadapterPath);
```

Comment:
As environment variables (`%temp%\adapters`) supported in runsettings file's "TestAdaptersPaths". Consider adding "Environment.ExpandEnvironmentVariables" here and test for same.

Reply:
@smadala, ok, it was fixed.